### PR TITLE
RZ spectral current correction and Galilean

### PR DESCRIPTION
**Pull request by dpgrote**
This is a big PR, implementing multiple new capabilities for the RZ spectral solver.

These are the main pieces of the changes:

- [x] Galilean PSATD solver
- [x] Current correction for both standard PSATD and Galilean PSATD
- [x] `fft_periodic_single_box` option (done in #1301)
- [x] Code rearranging and clean-up (done also in #1290)
- [x] Fix various bugs (done also in #1233, #1237, #1240, and #1287)
- [x] New CI tests: Langmuir_multi_rz_psatd_current_correction, galilean_rz_psatd, galilean_rz_psatd_current_correction

### CODE REVIEW

**Comment by EZoni** [comment]
@dpgrote Thank you for this PR! I will start setting up locally the tests that are needed and review the PR progressively as I address all the features that you introduced.

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+            amrex::Real const k_norm2 = kr*kr + kz*kz;
+
+            constexpr Complex I = Complex{0._rt,1._rt};
+
+            // Correct J
+            if ( k_norm2 != 0 )
+            {
+                Complex const theta2 = amrex::exp(I*kz*vz*dt);
+                Complex const inv_1_T2 = 1._rt/(kz*vz == 0 ?  1._rt : 1._rt - theta2);
+                Complex const j_corr_coef = (kz == 0 ? 1._rt/dt : I*kz*vz*inv_1_T2);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]: By looking at equations (C1)-(C2) of ([Lehe et al, PRE, 2016](https://journals.aps.org/pre/abstract/10.1103/PhysRevE.94.053305)), it seems to me that we should have a minus sign here. Could you crosscheck?
```suggestion
                Complex const j_corr_coef = (kz == 0 ? 1._rt/dt : -I*kz*vz*inv_1_T2);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+            Complex const X1 = X1_arr(i,j,k,mode);
+            Complex const X2 = X2_arr(i,j,k,mode);
+            Complex const X3 = X3_arr(i,j,k,mode);
+            Complex const X4 = X4_arr(i,j,k,mode);
+            Complex const T2 = Theta2_arr(i,j,k,mode);
+
+            // Update E (see WarpX online documentation: theory section)
+            fields(i,j,k,Ep_m) = T2*C*Ep_old
+                        + T2*S_ck*(-c2*I*kr/2._rt*Bz_old + c2*kz*Bp_old)
+                        + X4*Jp + kr*(X2*rho_new - T2*X3*rho_old);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]: By looking at equation (C4a) of ([Lehe et al, PRE, 2016](https://journals.aps.org/pre/abstract/10.1103/PhysRevE.94.053305)), it seems to me that we should have a factor 1/2 here. Could you crosscheck?
```suggestion
                        + X4*Jp + 0.5_rt*kr*(X2*rho_new - T2*X3*rho_old);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+            fields(i,j,k,Em_m) = T2*C*Em_old
+                        + T2*S_ck*(-c2*I*kr/2._rt*Bz_old - c2*kz*Bm_old)
+                        + X4*Jm - kr*(X2*rho_new - T2*X3*rho_old);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]: By looking at equation (C4b) of ([Lehe et al, PRE, 2016](https://journals.aps.org/pre/abstract/10.1103/PhysRevE.94.053305)), it seems to me that we should have a factor 1/2 here. Could you crosscheck?
```suggestion
                        + X4*Jm - 0.5_rt*kr*(X2*rho_new - T2*X3*rho_old);
```

**Comment by dpgrote** [comment]
Yes, the 1/2 is correct. This is the same correction that I had made to the non-Galilean version but forgot to carry it over to this version. Thanks for checking!

I’m working at a place that has a slow internet connection. Can you commit the fixes?

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp (hunk 1 above)
Comment (dpgrote) [reply Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]: Comparing to what is in fbpic, the minus sign is correct. Thanks for checking! I'll commit this change.

**Comment by EZoni** [comment]
Hi @dpgrote, could you fix the merge conflicts (due to #1246 and #1247 recently merged in `development`)?

**Comment by dpgrote** [comment]
@EZoni Merging done

**Comment by EZoni** [comment]
@dpgrote Since we merged #1301, could you merge `development` here again and fix the conflicting files? Thank you in advance!

**Comment by dpgrote** [comment]
@EZoni will do the merge...

**Comment by dpgrote** [comment]
@EZoni Merge is done

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/Make.package [hunk 4]
```diff
@@ -7,6 +7,7 @@ CEXE_sources += PMLPsatdAlgorithm.cpp
 ifeq ($(USE_RZ),TRUE)
   CEXE_sources += SpectralBaseAlgorithmRZ.cpp
   CEXE_sources += PsatdAlgorithmRZ.cpp
+  CEXE_sources += GalileanPsatdAlgorithmRZ.cpp
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/Make.package:10]: Could you add the corresponding entry in Source/FieldSolver/SpectralSolver/SpectralAlgorithms/CMakeLists.txt, in order to build with CMake?

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 5]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+ * \param[in,out] current two-dimensional array of unique pointers to MultiFab
+ *                        storing the three components of the current density
+ * \param[in]     rho     unique pointer to MultiFab storing the charge density
+ */
 void
 SpectralSolverRZ::CurrentCorrection (std::array<std::unique_ptr<amrex::MultiFab>,3>& current,
                                       const std::unique_ptr<amrex::MultiFab>& rho) {
      algorithm->CurrentCorrection(field_data, current, rho);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 6]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+/**
+  * \brief Public interface to call the member function ComputeSpectralDivE
+  * of the base class SpectralBaseAlgorithmRZ from objects of class SpectralSolverRZ
+  */
 void
 SpectralSolverRZ::ComputeSpectralDivE (const std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield,
                                        amrex::MultiFab& divE) {
     algorithm->ComputeSpectralDivE(field_data, Efield, divE);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 7]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+/* \brief Update the fields in spectral space, over one timestep */
 void
 SpectralSolverRZ::pushSpectralFields () {
     WARPX_PROFILE("SpectralSolverRZ::pushSpectralFields");
     // Virtual function: the actual function used here depends
     // on the sub-class of `SpectralBaseAlgorithm` that was
     // initialized in the constructor of `SpectralSolverRZ`
     algorithm->pushSpectralFields(field_data);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 8]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+/* \brief Transform spectral fields specified by `field_index1` and `field_index2`
+ * back to real space, and store it in `field_mf1` and `field_mf2`*/
 void
 SpectralSolverRZ::BackwardTransform (amrex::MultiFab& field_mf1, int const field_index1,
                                      amrex::MultiFab& field_mf2, int const field_index2) {
     WARPX_PROFILE("SpectralSolverRZ::BackwardTransform");
     field_data.BackwardTransform(field_mf1, field_index1,
                                  field_mf2, field_index2);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 9]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+ * real space, and store it in the component `i_comp` of `field_mf` */
 void
-SpectralSolverRZ::BackwardTransform (amrex::MultiFab& field_mf,
-                                     int const field_index,
+SpectralSolverRZ::BackwardTransform (amrex::MultiFab& field_mf, int const field_index,
                                      int const i_comp) {
     WARPX_PROFILE("SpectralSolverRZ::BackwardTransform");
     field_data.BackwardTransform(field_mf, field_index, i_comp);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 10]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+ *  to spectral space, and store the corresponding results internally
+ *  (in the spectral field specified by `field_index1` and `field_index2`) */
 void
 SpectralSolverRZ::ForwardTransform (amrex::MultiFab const & field_mf1, int const field_index1,
                                     amrex::MultiFab const & field_mf2, int const field_index2) {
     WARPX_PROFILE("SpectralSolverRZ::ForwardTransform");
     field_data.ForwardTransform(field_mf1, field_index1,
                                 field_mf2, field_index2);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp [hunk 11]
```diff
@@ -41,68 +43,97 @@ SpectralSolverRZ::SpectralSolverRZ (amrex::BoxArray const & realspace_ba,
...
+ *  (in the spectral field specified by `field_index`) */
 void
-SpectralSolverRZ::ForwardTransform (amrex::MultiFab const & field_mf,
-                                    int const field_index,
+SpectralSolverRZ::ForwardTransform (amrex::MultiFab const & field_mf, int const field_index,
                                     int const i_comp) {
     WARPX_PROFILE("SpectralSolverRZ::ForwardTransform");
     field_data.ForwardTransform(field_mf, field_index, i_comp);
-}
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralSolverRZ.cpp]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+class GalileanPsatdAlgorithmRZ : public SpectralBaseAlgorithmRZ
+{
+
+    public:
+        GalileanPsatdAlgorithmRZ(SpectralKSpaceRZ const & spectral_kspace,
+                                 amrex::DistributionMapping const & dm,
+                                 int const n_rz_azimuthal_modes, int const norder_z,
+                                 bool const nodal,
+                                 const amrex::Array<amrex::Real,3>& v_galilean,
+                                 amrex::Real const dt_step);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]:
```suggestion
        GalileanPsatdAlgorithmRZ (SpectralKSpaceRZ const & spectral_kspace,
                                  amrex::DistributionMapping const & dm,
                                  int const n_rz_azimuthal_modes, int const norder_z,
                                  bool const nodal,
                                  const amrex::Array<amrex::Real,3>& v_galilean,
                                  amrex::Real const dt_step);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        // Redefine functions from base class
+        virtual void pushSpectralFields(SpectralFieldDataRZ & f) override final;
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]:
```suggestion
        virtual void pushSpectralFields (SpectralFieldDataRZ & f) override final;
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+        virtual int getRequiredNumberOfFields() const override final {
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]:
```suggestion
        virtual int getRequiredNumberOfFields () const override final {
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+            return SpectralFieldIndex::n_fields;
+        }
+
+        void InitializeSpectralCoefficients(SpectralFieldDataRZ const & f);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]:
```suggestion
        void InitializeSpectralCoefficients (SpectralFieldDataRZ const & f);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+        /**
+         * \brief Virtual function for current correction in Fourier space
+         * (equation (19) of https://doi.org/10.1016/j.jcp.2013.03.010).
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]: I think this reference was copied from the Cartesian docs and does not really point to the RZ equations:
```suggestion
         * \brief Virtual function for current correction in Fourier space.
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+         *                           storing the three components of the current density
+         * \param[in]     rho        unique pointer to MultiFab storing the charge density
+         */
+        virtual void CurrentCorrection ( SpectralFieldDataRZ& field_data,
+                                         std::array<std::unique_ptr<amrex::MultiFab>,3>& current,
+                                         const std::unique_ptr<amrex::MultiFab>& rho ) override final;
+
+        /**
+         * \brief Virtual function for Vay current deposition in Fourier space
+         * (<a href="https://doi.org/10.1016/j.jcp.2013.03.010"> Vay et al, 2013</a>).
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.H]: Same, I think this reference was copied from the Cartesian docs and does not really point to the RZ equations:
```suggestion
         * \brief Virtual function for Vay current deposition in Fourier space.
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 18]
```diff
@@ -0,0 +1,356 @@
...
+                                                    int const n_rz_azimuthal_modes, int const norder_z,
+                                                    bool const nodal,
+                                                    const amrex::Array<amrex::Real,3>& v_galilean,
+                                                    amrex::Real const dt)
+     // Initialize members of base class
+     : SpectralBaseAlgorithmRZ(spectral_kspace, dm,
+                               norder_z, nodal),
+       m_dt(dt)
+{
+    m_v_galilean = v_galilean;
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]: Would it be possible to use member initialization for `m_v_galilean` as well?
```suggestion
     : SpectralBaseAlgorithmRZ(spectral_kspace, dm, norder_z, nodal),
       m_dt(dt),
       m_v_galilean(v_galilean)
{
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 19]
```diff
@@ -0,0 +1,356 @@
...
+                        + X1*(-I*kr/2._rt*Jz + kz*Jp);
+            fields(i,j,k,Bm_m) = T2*C*Bm_old
+                        - T2*S_ck*(-I*kr/2._rt*Ez_old - kz*Em_old)
+                        + X1*(-I*kr/2._rt*Jz - kz*Jm);
+            fields(i,j,k,Bz_m) = T2*C*Bz_old
+                        - T2*S_ck*I*(kr*Ep_old + kr*Em_old)
+                        + X1*I*(kr*Jp + kr*Jm);
+        });
+    }
+};
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp:154]: warning: extra ‘;’ [-Wpedantic]
```suggestion
}
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 20]
```diff
@@ -0,0 +1,356 @@
...
+            Complex const I = Complex{0._rt,1._rt};
+
+            // Calculate norm of vector
+            int const ir = i + nr*mode;
+            amrex::Real const kr = kr_arr[ir];
+            amrex::Real const kz = modified_kz[j];
+            amrex::Real const k_norm = std::sqrt(kr*kr + kz*kz);
+
+            // Calculate coefficients
+            if (k_norm != 0){
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
            if (k_norm != 0._rt){
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 21]
```diff
@@ -0,0 +1,356 @@
...
+                amrex::Real const kv = kz*vz;
+
+                amrex::Real const nu = kv/(k_norm*c);
+                Complex const theta = amrex::exp( 0.5_rt*I*kv*dt );
+                Complex const theta_star = amrex::exp( -0.5_rt*I*kv*dt );
+                Complex const e_theta = amrex::exp( I*c*k_norm*dt );
+
+                Theta2(i,j,k,mode) = theta*theta;
+
+                if ( (nu != 1.) && (nu != 0) ) {
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
                if ( (nu != 1._rt) && (nu != 0._rt) ) {
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 22]
```diff
@@ -0,0 +1,356 @@
...
+            amrex::Real const kz = modified_kz_arr[j];
+            amrex::Real const k_norm2 = kr*kr + kz*kz;
+
+            constexpr Complex I = Complex{0._rt,1._rt};
+
+            // Correct J
+            if ( k_norm2 != 0 )
+            {
+                Complex const theta2 = amrex::exp(I*kz*vz*dt);
+                Complex const inv_1_T2 = 1._rt/(kz*vz == 0 ?  1._rt : 1._rt - theta2);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
                Complex const inv_1_T2 = 1._rt/(kz*vz == 0._rt ?  1._rt : 1._rt - theta2);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 23]
```diff
@@ -0,0 +1,356 @@
...
+                Complex const j_corr_coef = (kz == 0 ? 1._rt/dt : -I*kz*vz*inv_1_T2);
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
                Complex const j_corr_coef = (kz == 0._rt ? 1._rt/dt : -I*kz*vz*inv_1_T2);
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 24]
```diff
@@ -0,0 +1,356 @@
...
+            // The k values for each mode are grouped together
+            int const ir = i + nr*mode;
+            amrex::Real const kr = kr_arr[ir];
+            amrex::Real const kz = modified_kz_arr[j];
+            amrex::Real const k_norm2 = kr*kr + kz*kz;
+
+            constexpr Complex I = Complex{0._rt,1._rt};
+
+            // Correct J
+            if ( k_norm2 != 0 )
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
            if ( k_norm2 != 0._rt )
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 25]
```diff
@@ -0,0 +1,356 @@
...
+                    X1(i,j,k,mode) = theta*x1/(ep0*c*c*k_norm*k_norm);
+                    // The difference betwen X2 and X3 below, and those
+                    // from the original paper is the factor ep0*k_norm*k_norm
+                    X2(i,j,k,mode) = (x1 - theta*(1._rt - C(i,j,k,mode)))
+                                     /(theta_star-theta)/(ep0*k_norm*k_norm);
+                    X3(i,j,k,mode) = (x1 - theta_star*(1._rt - C(i,j,k,mode)))
+                                     /(theta_star-theta)/(ep0*k_norm*k_norm);
+                    X4(i,j,k,mode) = I*kv*X1(i,j,k,mode) - theta*theta*S_ck(i,j,k,mode)/ep0;
+
+                } else if (nu == 0.) {
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
                } else if (nu == 0._rt) {
```

File: Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp [hunk 26]
```diff
@@ -0,0 +1,356 @@
...
+
+                    X1(i,j,k,mode) = (1._rt - C(i,j,k,mode))/(ep0 * c*c * k_norm*k_norm);
+                    X2(i,j,k,mode) = (1._rt - S_ck(i,j,k,mode)/dt)/(ep0 * k_norm*k_norm);
+                    X3(i,j,k,mode) = (C(i,j,k,mode) - S_ck(i,j,k,mode)/dt)/(ep0 * k_norm*k_norm);
+                    X4(i,j,k,mode) = -S_ck(i,j,k,mode)/ep0;
+
+                } else if ( nu == 1.) {
```
Comment (EZoni) [inline Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp]:
```suggestion
                } else if ( nu == 1._rt) {
```

**Review by EZoni — CHANGES_REQUESTED** [review]
@dpgrote Thanks again for this PR. I added some minor suggestions (mainly fixes of compiler warnings, a few `_rt` suffixes for single precision runs, and style). Overall, I think this is ready to merge. We will investigate possible further issues with the boosted frame separately.

**Comment by dpgrote** [comment]
@EZoni Thanks for your comments! I implemented all of the suggested changes.

**Review by EZoni — APPROVED** [review]

**Pull request merged**